DXMainClient/DXGUI/Multiplayer/GameInformationPanel.cs

```diff
@@ -194,17 +194,28 @@ public void SetInfo(GenericHostedGame game)
 
             if (mapLoader != null)
             {
-                mapTexture = mapLoader.GameModeMaps.Find(m => m.Map.UntranslatedName.Equals(game.Map, StringComparison.InvariantCultureIgnoreCase) && m.Map.IsPreviewTextureCached())?.Map?.LoadPreviewTexture();
+                Map map = mapLoader.GameModeMaps.Find(m => m.Map.UntranslatedName.Equals(game.Map, StringComparison.InvariantCultureIgnoreCase))?.Map;
+
+                if (map != null)
+                {
+                    if (map.IsPreviewTextureAvailableAsFile())
+                    {
+                        mapTexture = map.LoadPreviewTexture();
+                        disposeTextures = true;
+                    }
+                    else
+                    {
+                        mapTexture = AssetLoader.TextureFromImage(mapLoader.MapTextureCacheManager.GetMapTextureIfAvailable(map));
```

Comment:
Does this mean the first show of the preview will always show null texture? When will it be updated? IMO there should be a callback that will update the info upon loading the map.

Reply:
Yes. Currently it will be updated the second time it is visited.

Will trying implementing a callback after I am available again (~weeks perhaps)